src/libraries/Hooks.sol

```diff
@@ -141,6 +143,7 @@ library Hooks {
     {
         bytes memory result = callHook(self, data);
 
+        // If this hook wasnt meant to return something, default to 0 delta
```

Comment:
So here do `BeforeSwapDelta.ZERO_DELTA` and below decode it into BeforeSwapDelta type?

Reply:
Cannot do this as callHookWithReturnDelta is shared by different hooks